### ISSUE TITLE
Windows Gitian descriptor needs fixes

### ISSUE THREAD

**Issue by JeremyRand** [namecoin/namecoin-core#37]
As far as I can tell, no one's tested the Windows Gitian descriptor.  It needs the remote URL changed, for one thing.  Not sure what else it needs.  (Note: this ticket is for the master branch.)

**Comment by JeremyRand** [comment]
I'm in the process of fixing up the Windows Gitian descriptor.  There appear to be bugs in upstream Bitcoin Core's NSIS script (the word "bitcoin" is erroneously hardcoded in lots of places) which cause trouble for us.

**Comment by JeremyRand** [comment]
I got the Windows Gitian build to work; tested the 32-bit installer on WINE and everything looks okay.  It's still using Bitcoin graphics, but has the text "Namecoin".  I'm holding off on submitting a PR to Namecoin Core, because some of my changes might fit better as a PR to upstream Bitcoin Core.  I'll let you know when I hear back from the Bitcoin Core devs.

**Comment by JeremyRand** [comment]
Turns out that at least a subset of my changes were already submitted in a PR for Bitcoin master, so I'll wait for that to land in Bitcoin master before I do anything to Namecoin master.  This leaves open the question of what to do for the Namecoin 0.12 branch.  The Bitcoin master PR is more invasive than I would like to see for 0.12.  I can probably produce a minimalist patch for the 0.12 branch that will get Namecoin Core's .zip distributions to build.  I can also make the unsigned .exe installers build in 0.12 (also a minimalist patch), but they still will have Bitcoin graphics (although the Namecoin text should be correct).

Should we be introducing Namecoin graphics replacements for the Bitcoin installer graphics this late in the 0.12 branch cycle, or should we delay Windows installer releases until the Namecoin 0.13 release?  FWIW, it looks like substituting Namecoin graphics will be relatively easy; it's basically a matter of replacing some files.  I don't do graphics stuff much though.

**Comment by JeremyRand** [comment]
At the IRC dev meeting today, we agreed that Windows Gitian fixes should land in the 0.12 branch.  The master branch can wait for the upstream Bitcoin Core devs to decide what they want to do.  Any objections if I submit PR's for fixing Windows Gitian issues to the 0.12 branch?

@domob1812 @phelixbtc

**Comment by domob1812** [comment]
No objections.

**Comment by JeremyRand** [comment]
I've submitted a pull request to upstream Bitcoin that should fix the NSIS problems; I'll fix our Windows Gitian descriptor for master branch once that pull request gets merged.

**Comment by JeremyRand** [comment]
This should be working fine in 0.12 branch now.  master branch is still blocked by the NSIS script for the master branch, which in turn is still blocked by a pull request in upstream Bitcoin Core.  I just addressed the peer review in that pull request today, so hopefully we'll see progress on this front soon.

**Comment by JeremyRand** [comment]
We could split this ticket into 0.12 and master tickets.  0.12 ticket (which would be closed) would remain in the nc0.12.0 milestone; master ticket would be deferred to the nc0.12.2 milestone.  This would prevent the master branch from blocking the nc0.12.0 milestone, which it isn't really critical for.  Opinions?

**Comment by JeremyRand** [comment]
> We could split this ticket into 0.12 and master tickets. 0.12 ticket (which would be closed) would remain in the nc0.12.0 milestone; master ticket would be deferred to the nc0.12.2 milestone. This would prevent the master branch from blocking the nc0.12.0 milestone, which it isn't really critical for. Opinions?

@domob1812 ?

**Comment by domob1812** [comment]
I'm no expert about Gitian, so I have no strong opinion.  But I think it is a good idea to not block 0.12 on this, particularly since the 0.12 situation is fixed anyway.

**Comment by JeremyRand** [comment]
Okay.  I'll give 48 hours for anyone else to object; if no one does, I'll split the ticket.

**Comment by JeremyRand** [comment]
> Okay. I'll give 48 hours for anyone else to object; if no one does, I'll split the ticket.

Done.
